packages/blockchain-wallet-v4-frontend/src/data/modules/profile/sagas.ts

```diff
@@ -261,13 +262,28 @@ export default ({ api, coreSagas, networks }) => {
   }
 
   const resetUserKyc = function* () {
-    const retailToken = yield call(generateRetailToken)
-    const userId = (yield select(selectors.core.kvStore.userCredentials.getUserId)).getOrElse(null)
-    const lifetimeToken = (yield select(
-      selectors.core.kvStore.userCredentials.getLifetimeToken
-    )).getOrFail()
-    if (userId) {
-      yield call(api.resetUserKyc, userId, lifetimeToken, retailToken)
+    try {
+      yield put(A.resetUserKycLoading())
+      yield call(coreSagas.kvStore.walletCredentials.fetchMetadataWalletCredentials)
+      const retailToken = yield call(generateRetailToken)
+      const userId = (yield select(selectors.core.kvStore.userCredentials.getUserId)).getOrElse(
+        null
+      )
+      const lifetimeToken = (yield select(
+        selectors.core.kvStore.userCredentials.getLifetimeToken
+      )).getOrFail()
```

Comment:
(minor) since this part is only needed in case that `userId` exists, you an easily move it inside try block